### PR TITLE
SAML post

### PR DESCRIPTION
**Pull request by brian-vendia**
Blog post outlining how to use SAML IdP with Vendia Share.

### CODE REVIEW

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+longDescription: >
+    Vendia customers may have already made investments in identity services to allow 
+date: '2021-11-02'
+authors:
+  - Brian McNamara
+categories:
+  - architecture
+---
+
+Vendia Share is a Universal Application platform that accelerates the development of multi-party applications. Each party, or participant, has access to their own node within a Universal Application (Uni), allowing them to control a number of settings. One of the settings available is the authentication mechanisms API clients can use to access their node.
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:13]:
```suggestion
Vendia Share is a Universal Application platform that accelerates the development of multi-party applications. Each party, or participant, has access to their own node within a Universal Application (Uni), allowing them to control a number of settings. One of the settings available is the authentication mechanisms participants can use to access their node programatically.
```

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+Vendia Share allows participants to authenticate node API requests using several methods:
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:15]: Any reason not to list all options here?

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+* Amazon IAM
+* Amazon Cognito user pools
+* Custom authorizers using AWS Lambda
+
+In practice, one node can use Amazon IAM, another could use an Amazon Cognito user pool, and another could use a custom authorizer - all while sharing the same consistent and permissioned view of data. In this post, we're going to focus on Amazon Cognito user pools and, specifically, how to take advantage of an existing SAML 2.0 identity store to authenticate API requests to Vendia Share.
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:20]:
```suggestion
In practice, one node can use Amazon IAM, another could use an Amazon Cognito user pool, and another could use a custom authorizer - all while sharing the same consistent and permissioned view of data. In this post, we'll focus on Amazon Cognito user pools and, specifically, how to take advantage of an existing SAML 2.0 identity store to authenticate API requests to Vendia Share.
```

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+We'll begin by looking at SAML and the value it affords application end-users and identity administrators.
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:22]:
```suggestion
We'll begin by looking at SAML and the value it affords Uni participants and identity administrators.
```

Comment (james-vendia) [reply posts/2021-11-02-authenticating-requests-using-saml.md:22]: Just for consistency with terms use previously.

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+> - AWS Identity (https://aws.amazon.com/identity/saml/)
+
+In short, SAML allows users to authenticate to an existing identity provider and use generated credentials to access a downstream service or application, thus enabling a single sign-on experience. End users have the benefit of not having to use yet another set of credentials when accessing applications or services. Administrators improve their ability to manage users across a fleet of services. Providing and revoking access to SAML-enabled services becomes a streamlined process, all from one source - _the IdP_.
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:29]:
```suggestion
In short, SAML allows users to authenticate to an existing identity provider and use generated credentials to access a downstream service or application, thus enabling a single sign-on experience. End users get the benefit of using a single set of credentials when accessing different applications or services. Administrators improve their ability to manage users across a fleet of services. Providing and revoking access to SAML-enabled services becomes a streamlined process, all from one source - _the IdP_.
```

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+Now that we have a high-level understanding of SAML and the value it provides, let's explore Cognito user pools and how they can work together with SAML and Vendia Share.
+
+# SAML + Cognito User Pools + Vendia Share
+
+A user pool is a user directory in Amazon Cognito. It allows users to sign in to web and mobile applications through Amazon Cognito. User pools can be configured to allow users to sign in through SAML IdPs.
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:35]:
```suggestion
A user pool is a user directory in Amazon Cognito. It allows users to sign in to web and mobile applications through Amazon Cognito. User pools can also be configured to allow users to sign in through SAML IdPs.
```

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+Many identity services, including Okta, Ping Identity, Auth0, and Microsoft Active Directory Federation Services (ADFS), can function as SAML IdPs and work seamlessly with Cognito user pools. Regardless of the IdP, Amazon Cognito will return a JWT token to your mobile and web applications after a successful login. [AWS has created the following diagram to illustrate the authentication flow](https://docs.aws.amazon.com/cognito/latest/developerguide/images/amazon-cognito-user-pools-saml-idp-auth-flow.png).
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:37]: Maybe links to each of the IdPs you mentioned for ease of nav?

Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:37]:
```suggestion
Many identity services, including Okta, Ping Identity, Auth0, and Microsoft Active Directory Federation Services (ADFS), can function as SAML IdPs and work seamlessly with Cognito user pools. Regardless of the IdP, Amazon Cognito will return a JWT token to your mobile and web applications after a successful login. [AWS created a diagram to illustrate the authentication flow](https://docs.aws.amazon.com/cognito/latest/developerguide/images/amazon-cognito-user-pools-saml-idp-auth-flow.png).
```

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+
+![SAML federation authentication flow](https://docs.aws.amazon.com/cognito/latest/developerguide/images/amazon-cognito-user-pools-saml-idp-auth-flow.png)
+
+Vendia Share allows API requests to be authorized with the JWT tokens issued by Cognito. There is no additional friction to authenticate users when storing or retrieving data in a Vendia Share node. End users can continue using the same credentials they are likely already using to access SaaS applications. For them, an application using Share as a back-end component is just that - another application.
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:41]:
```suggestion
Vendia Share allows API requests to be authorized with the JWT issued by Cognito. There is no additional friction to authenticate users when storing or retrieving data in a Vendia Share node. End users can continue using the same credentials they are likely already using to access SaaS applications. For them, an application using Share as a back-end component is just that - another application.
```

**Review by james-vendia — CHANGES_REQUESTED** [review]
Some minor suggestions inline

File: posts/2021-11-02-authenticating-requests-using-saml.md (hunk 2 above)
Comment (brian-vendia) [reply posts/2021-11-02-authenticating-requests-using-saml.md:15]: Nope - just a miss on my part. Addressed in latest push.

File: posts/2021-11-02-authenticating-requests-using-saml.md (hunk 7 above)
Comment (brian-vendia) [reply posts/2021-11-02-authenticating-requests-using-saml.md]: Good call - updated in latest push.

File: posts/2021-11-02-authenticating-requests-using-saml.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+
+Now that we have a high-level understanding of SAML and the value it provides, let's explore Cognito user pools and how they can work together with SAML and Vendia Share.
+
+# SAML + Cognito User Pools + Vendia Share
+
+A user pool is a user directory in Amazon Cognito. It allows users to sign in to web and mobile applications through Amazon Cognito. User pools can also be configured to allow users to sign in through SAML IdPs.
+
+Many identity services, including [Okta](https://www.okta.com/), [Ping Identity](https://www.pingidentity.com/), [Auth0](https://auth0.com/), and [Microsoft Active Directory Federation Services (ADFS)](https://docs.microsoft.com/en-us/windows-server/identity/active-directory-federation-services), can function as SAML IdPs and work seamlessly with Cognito user pools. Regardless of the IdP, Amazon Cognito will return a JWT token to your mobile and web applications after a successful login. [AWS created a diagram to illustrate the authentication flow](https://docs.aws.amazon.com/cognito/latest/developerguide/images/amazon-cognito-user-pools-saml-idp-auth-flow.png).
```
Comment (james-vendia) [inline posts/2021-11-02-authenticating-requests-using-saml.md:38]:
```suggestion
Many identity services, including [Okta](https://www.okta.com/), [Ping Identity](https://www.pingidentity.com/), [Auth0](https://auth0.com/), and [Microsoft Active Directory Federation Services (ADFS)](https://docs.microsoft.com/en-us/windows-server/identity/active-directory-federation-services), can function as SAML IdPs and work seamlessly with Cognito user pools. Regardless of the IdP, Amazon Cognito will return a JWT to your mobile and web applications after a successful login. [AWS created a diagram to illustrate the authentication flow](https://docs.aws.amazon.com/cognito/latest/developerguide/images/amazon-cognito-user-pools-saml-idp-auth-flow.png).
```

**Review by james-vendia — APPROVED** [review]
Approved, with one remaining (very minor) suggestion.

**Pull request merged**